libs/database/src/collab.rs

```diff
@@ -93,266 +91,118 @@ impl Default for StorageConfig {
 }
 
 #[derive(Clone)]
-pub struct CollabPostgresDBStorageImpl {
+pub struct CollabDatabaseStorageImpl {
   #[allow(dead_code)]
   pg_pool: PgPool,
+  redis_client: ConnectionManager,
   config: StorageConfig,
 }
 
 pub const FLUSH_PER_UPDATE: u32 = 100;
-impl CollabPostgresDBStorageImpl {
-  pub fn new(pg_pool: PgPool) -> Self {
+impl CollabDatabaseStorageImpl {
+  pub fn new(pg_pool: PgPool, redis_client: ConnectionManager) -> Self {
     let config = StorageConfig {
       flush_per_update: FLUSH_PER_UPDATE,
     };
-    Self { pg_pool, config }
+    Self {
+      pg_pool,
+      redis_client,
+      config,
+    }
   }
 }
 
 #[async_trait]
-impl CollabStorage for CollabPostgresDBStorageImpl {
+impl CollabStorage for CollabDatabaseStorageImpl {
```

Comment:
Create a new implementation of CollabStorage and keep the CollabPostgresDBStorageImpl.